pyiceberg/table/snapshots.py

```diff
@@ -249,6 +241,9 @@ class Snapshot(IcebergBaseModel):
     summary: Optional[Summary] = Field(default=None)
     schema_id: Optional[int] = Field(alias="schema-id", default=None)
 
+    # Private attribute for caching the manifests
+    _manifest_cache: Optional[List[ManifestFile]] = PrivateAttr(default=None)
```

Comment:
This looks promising @kevinjqliu 

I know this is a regression from the intended optimization at the module level as suggested here: https://github.com/apache/iceberg-python/pull/787#discussion_r1632224368 but I don't think we have a better alternative 🙂